check_api/src/main/java/com/google/errorprone/SuppressionInfo.java

```diff
@@ -61,11 +69,85 @@ public class SuppressionInfo {
 
   private final boolean inGeneratedCode;
 
+  @BugPattern(summary = "Warns when a @SuppressWarnings is not needed", severity = WARNING)
+  private static final class UnusedSuppressionChecker extends BugChecker {}
```

Comment:
Is it possible to suppress UnusedSuppressionChecker findings? This isn't a feature request to do that, I'm just thinking through it.

I suppose one reason to want that would be if an Error Prone check name shadowed a suppression string used by another tool, and Error Prone didn't report the findings but the other tool would.

Reply:
I guess I'm not sure about this.  On the one hand, I think it's probably generally good for all warnings to be suppressible somehow.  [Javac is working on adding a similar check](https://github.com/openjdk/jdk/pull/25167) and there you can suppress the unneeded suppression warning.

On the other hand, I wasn't sure if users would always run with this unused suppression check enabled.  That could potentially negatively impact build times, since without this new check, Error Prone completely skips running checkers on sub-trees where the warnings would be suppressed.  But maybe users don't rely on that?

In any case, I can add this support if we decide to do it.